File: practical/amstrong.java
package practical;
import java.util.*;
public class amstrong {
public static void main(String[] args) {
	Scanner sc=new Scanner(System.in);
	double sum=0;
	int r=0,cnt=0;
	System.out.println("Enter Number : ");
	int no=sc.nextInt();
	int a=no;
	while(no>0) {
		cnt++;
		no/=10;
	}
	System.out.println(cnt);
	no=a;
	while(no>0) {
		r=no%10;
		sum=sum+Math.pow(r,cnt);
		no/=10;
	}
	if(sum==a) {
		System.out.println("Amstrong");
	}
	else {
		System.out.println("Not Amstrong");
	}
	sc.close();
}
}

File: c/amstrong.c
#include <stdio.h>
#include <math.h>
int main()
{
	int n, s = 0, a, cnt = 0, x;
x:
	printf("enter number :");
	scanf("%d", &n);
	a = n;
	do
	{
		cnt++;
		n = n / 10;
	} while (n > 0);
	n = a;
	while (n > 0)
	{

		s = s + pow(n % 10, cnt);
		n = n / 10;
	}
	if (a == s)
		printf("%d is amstrong number \n", a);
	else
		printf("%d is not amstrong number \n", a);
	printf("\nfor continue press 1\n");
	scanf("%d", &x);
	if (x == 1)
	{
		goto x;
	}
	else
	{
	}
	return 1;
}
